src/common/ossl_helpers.c

```diff
@@ -53,6 +55,11 @@ void oqs_free_ossl_objects(void) {
 
 const EVP_MD *oqs_sha256(void) {
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	if (!sha256_ptr) {
```

Comment:
Strictly speaking this depends on access to the pointer being atomic. Other than that this should work fine.

Reply:
> Strictly speaking this depends on writes to the pointer being atomic. 

There's only a single function where this (or any of these) pointer is written: https://github.com/open-quantum-safe/liboqs/blob/e7bc261841a459631e6a389b0377e572b9e63b78/src/common/ossl_helpers.c#L16-L30 

And _that_ should be guarded via/accessible by only one thread via `CRYPTO_THREAD_run_once`, right?

So changing this to Ready For Review. @beldmit : As author of https://github.com/open-quantum-safe/liboqs/pull/1431 any further input from your side?

Reply:
Sorry for unclarity, I mean I wanted to avoid the synchronization/thread-safety stuff. The approach LGTM

Reply:
> There's only a single function where this (or any of these) pointer is written:
> 
> https://github.com/open-quantum-safe/liboqs/blob/e7bc261841a459631e6a389b0377e572b9e63b78/src/common/ossl_helpers.c#L16-L30

If the pointer is written non-atomically the check could load some half-written pointer which is non-null but also invalid, the invalid (non-NULL) pointer would be then returned from the function.

In reality I do not think it can happen apart from some truly exotic hardware/platforms.